pageserver/src/walredo.rs

```diff
@@ -66,22 +69,23 @@ pub fn wal_redo_main(conf: &PageServerConf, timelineid: ZTimelineId) {
             let _guard = runtime.enter();
             process = WalRedoProcess::launch(&datadir, &runtime).unwrap();
         }
+        info!("WAL redo postgres started");
 
         // Pretty arbitrarily, reuse the same Postgres process for 100 requests.
         // After that, kill it and start a new one. This is mostly to avoid
         // using up all shared buffers in Postgres's shared buffer cache; we don't
         // want to write any pages to disk in the WAL redo process.
-        for _i in 1..100 {
+        for _i in 1..100000 {
             let request = walredo_channel_receiver.recv().unwrap();
 
             let result = handle_apply_request(&pcache, &process, &runtime, request);
             if result.is_err() {
                 // On error, kill the process.
+                error!("Kill wal redo process on error");
                 break;
             }
```

Comment:
I don't really understand the comment and message.

Is this saying "in the future we should kill the process"?  "Somebody already killed the process?"

Reply:
Sorry, it is not my code. I think it is better to ask @hlinnaka  about his intention.

Reply:
I hope this clarifies it:
```
diff --git a/pageserver/src/walredo.rs b/pageserver/src/walredo.rs
index bf0159a..e46568f 100644
--- a/pageserver/src/walredo.rs
+++ b/pageserver/src/walredo.rs
@@ -76,11 +76,17 @@ pub fn wal_redo_main(conf: &PageServerConf, timelineid: ZTimelineId) {
 
             let result = handle_apply_request(&pcache, &process, &runtime, request);
             if result.is_err() {
-                // On error, kill the process.
+                // Something went wrong with handling the request. It's not clear
+                // if the request was faulty, and the next request would succeed
+                // again, or if the 'postgres' process went haywire. To be safe,
+                // kill the 'postgres' process so that we will start from a clean
+                // slate, with a new process, for the next request.
                 break;
             }
         }
 
+        // Time to kill the 'postgres' process. A new one will be launched on next
+        // iteration of the loop.
         info!("killing WAL redo postgres process");
         let _ = runtime.block_on(process.stdin.get_mut().shutdown());
         let mut child = process.child;
```